backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/GL_Journal_Builder.java

```diff
@@ -43,50 +41,41 @@
  * #L%
  */
 
-/**
- * {@link I_GL_Journal} and {@link I_GL_JournalLine} builder.
- * 
- * To create a new builder instance please use {@link #newBuilder(I_GL_JournalBatch)}.
- * 
- * @author metas-dev <dev@metasfresh.com>
- *
- */
 public class GL_Journal_Builder
 {
-	public static final GL_Journal_Builder newBuilder(final I_GL_JournalBatch glJournalBatch)
+	public static final GL_Journal_Builder newBuilder(final GLJournalRequest glJournalRequest)
 	{
-		return new GL_Journal_Builder(glJournalBatch);
+		return new GL_Journal_Builder(glJournalRequest);
 	}
 
 	// services
 	private final transient ICurrencyDAO currencyDAO = Services.get(ICurrencyDAO.class);
+	private final IGLJournalDAO glJournalDAO = Services.get(IGLJournalDAO.class);
 
-	private final I_GL_JournalBatch glJournalBatch;
 	private final I_GL_Journal glJournal;
 
 	private final List<GL_JournalLine_Builder> glJournalLineBuilders = new ArrayList<>();
 
-	private GL_Journal_Builder(final I_GL_JournalBatch glJournalBatch)
+	private GL_Journal_Builder(final GLJournalRequest request)
```

Comment:
makes no sense having this "partial" request here. 
Pls have all the required values in the request 
or introduce a setter method for each of them.

Reply:
Done. I chose to get rid of the setters and use the request.